internal/runbits/legacy/projectmigration/projectmigration.go

```diff
@@ -0,0 +1,79 @@
+package projectmigration
+
+import (
+	"path/filepath"
+
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/multilog"
+	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/internal/prompt"
+	"github.com/ActiveState/cli/pkg/localcommit"
+	"github.com/ActiveState/cli/pkg/projectfile"
+)
+
+type projecter interface {
+	Dir() string
+	URL() string
+	Path() string
+	LegacyCommitID() string
+}
+
+var prompter prompt.Prompter
+var out output.Outputer
+var declined bool
+
+// Register exists to avoid boilerplate in passing prompt and out to every caller of
+// commitmediator.Get() for retrieving legacy commitId from activestate.yaml.
+// This is an anti-pattern and is only used to make this legacy feature palatable.
+func Register(prompter_ prompt.Prompter, out_ output.Outputer) {
+	prompter = prompter_
+	out = out_
+}
+
+func PromptAndMigrate(proj projecter) (bool, error) {
+	if prompter == nil || out == nil {
+		return false, errs.New("projectmigration.Register() has not been called")
+	}
+
+	if declined {
+		return false, nil
+	}
+
+	defaultChoice := false
+	if migrate, err := prompter.Confirm("", locale.T("projectmigration_confirm"), &defaultChoice); err == nil && !migrate {
+		if out.Config().Interactive {
+			out.Notice(locale.Tl("projectmigration_declined", "Migration declined for now"))
+		}
+		declined = true
+		return false, nil
+	} else if err != nil {
+		return false, errs.Wrap(err, "Could not confirm migration choice")
+	}
+
+	if err := localcommit.Set(proj.Dir(), proj.LegacyCommitID()); err != nil {
+		return false, errs.Wrap(err, "Could not create local commit file")
+	}
+
+	if fileutils.DirExists(filepath.Join(proj.Dir(), ".git")) {
+		err := localcommit.AddToGitIgnore(proj.Dir())
+		if err != nil {
+			multilog.Error("Unable to add local commit file to .gitignore: %v", err)
```

Comment:
This would almost certainly be due to something client-side like permission errors I think? We should probably not log this to rollbar. Or maybe add a check to only log if it's not a permission or IO error.

Reply:
I wasn't able to find a generic ErrIO variable, but I did find ErrPermission, so we'll go with that.